adam/opm_params.py

```diff
@@ -302,4 +306,46 @@ def generate_opm(self):
+        anomaly_angle_cov = ""
+        if self._keplerian_covariance is not None:
+            if using_mean_anomaly:
+                anomaly_angle_cov = ("USER_DEFINED_CM_A = %s\n" % (self._keplerian_covariance[15])) + \
+                                    ("USER_DEFINED_CM_E = %s\n" % (self._keplerian_covariance[16])) + \
+                                    ("USER_DEFINED_CM_I = %s\n" % (self._keplerian_covariance[17])) + \
+                                    ("USER_DEFINED_CM_O = %s\n" % (self._keplerian_covariance[18])) + \
+                                    ("USER_DEFINED_CM_W = %s\n" % (self._keplerian_covariance[19])) + \
+                                    ("USER_DEFINED_CM_M = %s\n" % (self._keplerian_covariance[20]))
+            else:
+                anomaly_angle_cov = ("USER_DEFINED_CT_A = %s\n" % (self._keplerian_covariance[15])) + \
+                                    ("USER_DEFINED_CT_E = %s\n" % (self._keplerian_covariance[16])) + \
+                                    ("USER_DEFINED_CT_I = %s\n" % (self._keplerian_covariance[17])) + \
+                                    ("USER_DEFINED_CT_O = %s\n" % (self._keplerian_covariance[18])) + \
+                                    ("USER_DEFINED_CT_W = %s\n" % (self._keplerian_covariance[19])) + \
+                                    ("USER_DEFINED_CT_T = %s\n" % (self._keplerian_covariance[20]))
+
+            covariance = ("USER_DEFINED_CA_A = %s\n" % (self._keplerian_covariance[0])) + \
+                         ("USER_DEFINED_CE_A = %s\n" % (self._keplerian_covariance[1])) + \
+                         ("USER_DEFINED_CE_E = %s\n" % (self._keplerian_covariance[2])) + \
+                         ("USER_DEFINED_CI_A = %s\n" % (self._keplerian_covariance[3])) + \
+                         ("USER_DEFINED_CI_E = %s\n" % (self._keplerian_covariance[4])) + \
+                         ("USER_DEFINED_CI_I = %s\n" % (self._keplerian_covariance[5])) + \
+                         ("USER_DEFINED_CO_A = %s\n" % (self._keplerian_covariance[6])) + \
+                         ("USER_DEFINED_CO_E = %s\n" % (self._keplerian_covariance[7])) + \
+                         ("USER_DEFINED_CO_I = %s\n" % (self._keplerian_covariance[8])) + \
+                         ("USER_DEFINED_CO_O = %s\n" % (self._keplerian_covariance[9])) + \
+                         ("USER_DEFINED_CW_A = %s\n" % (self._keplerian_covariance[10])) + \
+                         ("USER_DEFINED_CW_E = %s\n" % (self._keplerian_covariance[11])) + \
+                         ("USER_DEFINED_CW_I = %s\n" % (self._keplerian_covariance[12])) + \
+                         ("USER_DEFINED_CW_O = %s\n" % (self._keplerian_covariance[13])) + \
+                         ("USER_DEFINED_CW_W = %s\n" % (self._keplerian_covariance[14])) + \
+                         anomaly_angle_cov
+
+        return base_opm + keplerian_elements + spacecraft_params + covariance + maneuver + keplerian_covariance
+
+    def check_params(self, allowed, actual):
```

Comment:
nit: this is a public method, should the name be a little more specific? e.g. check_for_extra_opm_params or something

Reply:
It should be a private method. Let me change that